Makefile

```diff
@@ -27,93 +27,97 @@ NAME = lib50-c
+	gcc -o "$(LIB_DIR)/libcs50.so" -shared "$(OBJ)"
+	rm -f "$(OBJ)"
+	cp "$(HDR)" "$(INCLUDE_DIR)"
+	cp "$(SRC)" "$(BUILD_SRC)"
+	cp -r docs/* "$(MAN_DIR)"
+	find "$(BUILD_DIR)" -type d -exec chmod 0755 {} +
+	find "$(BUILD_DIR)" -type f -exec chmod 0644 {} +
 
 .PHONY: clean
 clean:
-	rm -rf $(BUILD_DIR)
+	rm -rf "$(BUILD_DIR)"
 
 .PHONY: deb
 deb: build
 	fpm \
-	-C $(BUILD_DIR) \
+	-C "$(BUILD_DIR)" \
 	-m "$(MAINTAINER)" \
 	-n "$(NAME)" \
-	-p $(BUILD_DIR) \
+	-p "$(BUILD_DIR)" \
 	-s dir \
 	-t deb \
-	-v $(VERSION) \
+	-v "$(VERSION)" \
 	--conflicts "$(NAME) (<< $(VERSION))" \
-	--conflicts $(OLD_NAME) \
+	--conflicts "$(OLD_NAME)" \
 	--deb-no-default-config-files \
 	--depends c-compiler \
 	--description "$(DESCRIPTION)" \
-	--provides $(NAME) \
-	--provides $(OLD_NAME) \
+	--provides "$(NAME)" \
+	--provides "$(OLD_NAME)" \
 	--replaces "$(NAME) (<= $(VERSION))" \
-	--replaces $(OLD_NAME) \
+	--replaces "$(OLD_NAME)" \
 	usr
 
 .PHONY: hackerrank
 hackerrank: build
-	cat $(HDR) > $(HR_HDR)
-	echo "\n#ifndef _CS50_C\n#define _CS50_C\n" >> $(HR_HDR)
-	cat $(SRC) >> $(HR_HDR)
-	echo "\n#endif" >> $(HR_HDR)
+	cat "$(HDR)" > "$(HR_HDR)"
+	echo "\n#ifndef _CS50_C\n#define _CS50_C\n" >> "$(HR_HDR)"
+	cat "$(SRC)" >> "$(HR_HDR)"
+	echo "\n#endif" >> "$(HR_HDR)"
 
 .PHONY: install
 install: build
-	cp $(INCLUDE_DIR)/* /usr/include
-	cp $(LIB_DIR)/* /usr/lib
+	cp "$(INCLUDE_DIR)/*" /usr/include
+	cp "$(LIB_DIR)/*" /usr/lib
 
 # TODO: add dependencies
 .PHONY: pacman
 pacman: build
-	rm -f $(NAME)-$(VERSION)-*.pkg.tar.xz
+	rm -f "$(NAME)-$(VERSION)-*.pkg.tar.xz"
 	fpm \
-	-C $(BUILD_DIR) \
+	-C "$(BUILD_DIR)" \
 	-m "$(MAINTAINER)" \
 	-n "$(NAME)" \
-	-p $(BUILD_DIR) \
+	-p "$(BUILD_DIR)" \
 	-s dir \
 	-t pacman \
-	-v $(VERSION) \
+	-v "$(VERSION)" \
 	--description "$(DESCRIPTION)" \
 	usr
 
 # TODO: add dependencies
 .PHONY: rpm
 rpm: build
-	rm -f $(NAME)-$(VERSION)-*.rpm
+	rm -f "$(NAME)-$(VERSION)-*.rpm"
 	fpm \
-	-C $(BUILD_DIR) \
+	-C "$(BUILD_DIR)" \
 	-m "$(MAINTAINER)" \
 	-n "$(NAME)" \
-	-p $(BUILD_DIR) \
+	-p "$(BUILD_DIR)" \
 	-s dir \
 	-t rpm \
-	-v $(VERSION) \
+	-v "$(VERSION)" \
 	--description "$(DESCRIPTION)" \
 	usr
 
 # TODO: improve test suite
 .PHONY: test
 test: build hackerrank
-	clang -ggdb3 -I$(INCLUDE_DIR) -O0 -std=c99 -Wall -Werror -Wno-deprecated-declarations $(TESTS_DIR)/eprintf.c -L$(LIB_DIR) -lcs50 -o $(EPRINTF_EXE)
-	clang -I$(BUILD_DIR) -std=c99 -Wall -Werror -Wno-deprecated-declarations $(TESTS_DIR)/hackerrank.c -o $(HR_EXE)
-	LD_LIBRARY_PATH=$(LIB_DIR) $(EPRINTF_EXE)
-	$(HR_EXE)
+	clang -ggdb3 -I "$(INCLUDE_DIR)" -O0 -std=c99 -Wall -Werror -Wno-deprecated-declarations "$(TESTS_DIR)/eprintf.c" -L "$(LIB_DIR)" -lcs50 -o "$(EPRINTF_EXE)"
+	clang -I "$(BUILD_DIR)" -std=c99 -Wall -Werror -Wno-deprecated-declarations "$(TESTS_DIR)/hackerrank.c" -o "$(HR_EXE)"
+	LD_LIBRARY_PATH="$(LIB_DIR)" $(subst $(SPACE),\ ,$(EPRINTF_EXE))
+	$(subst $(SPACE),\ ,$(HR_EXE))
```

Comment:
Interesting! Seems to work. Should've tried, but assumed it wouldn't work since normally `"foo"`, where `foo` is executable doesn't. Turns out `"./foo"` works :neutral_face:.

Replaced per beadbaf. Thanks!